### DISCUSSION TITLE
MongoDB ACL Cache Configuration performance

### DISCUSSION THREAD

**Discussion by ctellechea2001** [emqx/emqx#11220] [Q&A] [answered]
### What happened?

Hello, how are you? I need your help to configure ACL cache correctly.

Currently we are using mongodb as ACL. We don't have ACL cache configured in the EMQX.

Last month we saw that the queries to mqtt_acl collections is taking more cpu than normal (The quantity of devices connected of emqx has been increased too).

![imagen](https://github.com/emqx/emqx/assets/45426180/ea88ac5a-c60a-496c-9c0a-d232600408c7)

We are planning to implement ACL Cache to get best performance on mongodb atlas.

What values do you recommed to implement of **Maximum numbers of cache items** and **time to live for the cached data**?

We are using emqx 5.0.19 on lower enviroments and 4.3.8 in the prd enviroment.

![imagen](https://github.com/emqx/emqx/assets/45426180/06f10c5b-3d0d-434d-98c7-5653471a481e)

### What did you expect to happen?

Improve better mongo atlas performance using ACL cache.

### How can we reproduce it (as minimally and precisely as possible)?

N/A

### Anything else we need to know?

N/A

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
# paste output here
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Log files

<details>

</details>

**Comment by SergeTupchiy** [comment, marked as answer]
Hi @ctellechea2001,
I suggest starting with the defaults values of 32/1min. Please note that authorization cache works per client (connection) basis: it improves the performance when the client publishes and subscribes frequently. In other words, it can't reduce load on an external authorization provider (MongoDB in your case) when a large number of new connections appears.
